File: knot-server/knot-service/src/main/java/org/chobit/knot/gateway/vo/routing/RoutingRule.java
package org.chobit.knot.gateway.vo.routing;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import org.chobit.knot.gateway.model.QuotaPolicy;
import org.chobit.knot.gateway.model.RateLimitPolicy;

import java.util.List;

public record RoutingRule(
        Long id,
        @Size(max = 32) String ruleCode,
        @NotBlank String name,
        @Size(max = 128) String appScenario,
        List<@Size(max = 64) String> modelTypes,
        List<Long> consumerIds,
        List<String> consumerNames,
        Long appId,
        String appName,
        Long userId,
        String userName,
        @NotBlank String strategy,
        boolean enabled,
        @Valid List<RoutingRuleModelItem> models,
        RateLimitPolicy rateLimitPolicy,
        QuotaPolicy quotaPolicy
) {
}

File: knot-front/src/utils/routingRule.js
/** 生成路由规则编码（32 位十六进制，与后端 RoutingRuleCodeGenerator 一致） */
export function generateRoutingRuleCode() {
  return crypto.randomUUID().replaceAll("-", "");
}

/** 构建路由规则更新载荷（含启用开关切换） */
export function buildRoutingRulePayload(row, enabled) {
  return {
    ruleCode: row.ruleCode,
    name: row.name,
    appScenario: row.appScenario ?? null,
    modelTypes: Array.isArray(row.modelTypes) && row.modelTypes.length ? row.modelTypes : ["CHAT"],
    consumerIds: Array.isArray(row.consumerIds) ? row.consumerIds : [],
    appId: row.appId,
    userId: row.userId ?? null,
    strategy: row.strategy,
    enabled,
    models: (row.models || []).map((m) => ({
      modelId: m.modelId,
      priority: m.priority ?? 100,
      primary: !!m.primary
    })),
    rateLimitPolicy: row.rateLimitPolicy ?? null,
    quotaPolicy: row.quotaPolicy ?? null
  };
}
